modules/scripts/startup-script/main.tf

```diff
@@ -89,9 +88,9 @@ resource "google_storage_bucket" "configs_bucket" {
 resource "google_storage_bucket_object" "scripts" {
   # this writes all scripts exactly once into GCS
   for_each = local.runners_map
-  name     = "${local.storage_folder_path_prefix}${each.key}"
-  content  = each.value["content"]
-  source   = each.value["source"]
+  name     = "${local.storage_folder_path_prefix}${each.key}-${try(md5(each.value.content), filemd5(each.value.source))}"
```

Comment:
This will have the effect of replacing every deployed startup script file on a version update re-apply. This in turn will update the master startup script, which reference the file names in buckets.

I think generally this won't be very impactful. It will result in update to metadata on VMs and on restart will run the new scripts fine. In the case of Batch it will force replacement of instance template, which I also don't think is a big deal.

So overall, let's move forward but just think it is worth pointing out, especially if you can think of some case I am missing. Feel free to close this comment.

Reply:
If what you mean is "upon the next release every customer will experience a destroy/create" for every object managed by the startup-script module, even when the content has not changed, yes. It's essentially an unavoidable consequence of being chained to a create_before_destroy resource. But, that will propagate correctly as you describe and should be noted clearly in the release notes.